crates/flox/src/commands/general.rs

```diff
@@ -132,9 +146,48 @@ pub enum GeneralCommands {
     ResetMetrics,
 
     /// access to the nix CLI
+    #[bpaf(command)]
     Nix(#[bpaf(external(parse_nix_passthru))] WrappedNix),
 }
 
+#[allow(unused)] // pending rust implementation
+#[derive(Bpaf, Clone)]
+pub struct LoginArgs {
+    /// the hostname of the flox hub to authenticate with
+    #[bpaf(long, short, argument("hostname"))]
+    hostname: String,
```

Comment:
I guess that is optional, right?

```suggestion
    hostname: Option<String>,
```